ext/mysqli/mysqli_prop.c

```diff
@@ -71,38 +71,59 @@ if (!obj->ptr) { \
 	p = (MYSQL_STMT *)((MY_STMT *)((MYSQLI_RESOURCE *)(obj->ptr))->ptr)->stmt; \
 }
 
-#define MYSQLI_MAP_PROPERTY_FUNC_LONG( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
+#define MYSQLI_MAP_PROPERTY_FUNC_LONG_OR_STR( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
 static int __func(mysqli_object *obj, zval *retval, bool quiet) \
 {\
 	__ret_type l;\
 	__get_type;\
-	if (!p) {\
-		ZVAL_NULL(retval);\
-	} else {\
-		l = (__ret_type)__int_func(p);\
-		if (l < ZEND_LONG_MAX) {\
-			ZVAL_LONG(retval, (zend_long) l);\
-		} else { \
-			ZVAL_NEW_STR(retval, strpprintf(0, __ret_type_sprint_mod, l)); \
-		} \
+	ZEND_ASSERT(p);\
+	l = (__ret_type)__int_func(p);\
+	if (l < ZEND_LONG_MAX) {\
+		ZVAL_LONG(retval, (zend_long) l);\
+	} else { \
+		ZVAL_NEW_STR(retval, strpprintf(0, __ret_type_sprint_mod, l)); \
 	} \
 	return SUCCESS; \
 }
 
-#define MYSQLI_MAP_PROPERTY_FUNC_STRING(__func, __int_func, __get_type)\
+#define MYSQLI_MAP_PROPERTY_FUNC_LONG( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
+static int __func(mysqli_object *obj, zval *retval, bool quiet) \
+{\
+	__ret_type l;\
+	__get_type;\
+	ZEND_ASSERT(p);\
+	l = (__ret_type)__int_func(p);\
+	ZEND_ASSERT(l < ZEND_LONG_MAX);\
+	ZVAL_LONG(retval, (zend_long) l);\
+	return SUCCESS; \
+}
+
+#define MYSQLI_MAP_PROPERTY_FUNC_STR_OR_NULL(__func, __int_func, __get_type)\
 static int __func(mysqli_object *obj, zval *retval, bool quiet)\
 {\
 	char *c;\
 	__get_type;\
-	if (!p) {\
+	ZEND_ASSERT(p);\
+	c = (char *)__int_func(p);\
+	if (c) {\
+		ZVAL_STRING(retval, c);\
+	} else {\
 		ZVAL_NULL(retval);\
+	}\
+	return SUCCESS; \
+}
+
+#define MYSQLI_MAP_PROPERTY_FUNC_STR(__func, __int_func, __get_type)\
+static int __func(mysqli_object *obj, zval *retval, bool quiet)\
+{\
+	char *c;\
+	__get_type;\
+	ZEND_ASSERT(p);\
+	c = (char *)__int_func(p);\
+	if (c) {\
```

Comment:
Make this `ZEND_ASSERT(c)` instead of branch + `ZEND_UNREACHABLE()`.

Reply:
Thank you! :) And thanks for the help, @kamil-tekiela! This one sneaked in from earlier, I'll fix it while applying the change.